### PR TITLE
chore(issue-110): move all dev dependencies to dev-packages

### LINKED ISSUE #200: Designate DevDependencies BE
Certain packages and libraries are only needed when actively developing the project, and are unnecessary when deployed. For example, testing suites are important to make sure covered code is behaving properly, but have no effect on how the project runs. These are known as DevDependencies.

DevDependencies should first be identified. If they are not already marked as such in the Pipfile or package.json, uninstall then reinstall the package using one of the following commands:

Backend: pipenv install --dev [package-name]

Frontend: npm install [package-name] --save-dev

### PR DESCRIPTION
**Pull request by esparr**
### **1. Targeted Issue** ###

(Issue #110)[https://trello.com/c/MZOUTyBS]

### **2. Overview of Solution** ###

Moves the following development packages to `[dev-packages]` in Pipfile
- coverage
- pre-commit

Adds documentation to README.md about installing packages.

### **3. Tools Used** ###

Pipenv category groups - https://pipenv.pypa.io/en/latest/pipfile/#package-category-groups

### **4. Testing Strategy** ###

**We will need to test this on staging**

### **5. Future Implications** ###

Moving forward we should make sure we add dev-packages using the following command: 
`pipenv install --dev <package-name>`

### **6. Screenshots** ###

Screen Shot 2023-11-02 at 1 40 30 PM

### **7. Code Reviewers** ###

@GitLukeW @jimbandrews 

---

Before you go...

- [x]  I have linked the relevant issue
- [x]  I have provided an overview of my solution
- [x]  I have shared the tools I've used
- [x]  I have outlined my testing strategy
- [x]  I have considered future implications of my changes
- [x]  I have included relevant screenshots
- [x]  I have kept my team in the loop

### CODE REVIEW

**Review by jimbandrews — APPROVED** [review]
New doc section looks good!

File: README.md [hunk 1]
```diff
@@ -267,6 +268,29 @@ As long as your are running locally, use the value `dev`.
 
 3. Save the .env file.
 
+
+## Adding Packages
+
+(Pipenv)[https://pipenv.pypa.io/en/latest/] is a packaging tool for Python that solves some common problems associated with the typical development workflow using pip, virtualenv, and the requirements.txt. In addition to addressing some common issues, it consolidates and simplifies the development process to a single command line tool.
```
Comment (GitLukeW) [inline README.md]: Looks like the () and [] are flipped flopped here. Should be [Pipenv](https://pipenv.pypa.io/en/latest/)

File: README.md [hunk 2]
```diff
@@ -267,6 +268,29 @@ As long as your are running locally, use the value `dev`.
...
+```bash
+pipenv install <package-name>
+```
+
+If the package is only needed for development, please run installation as follows:
+
+```bash
+pipenv install <package-name> --dev
+```
+Providing the --dev argument will put the dependency in a special [dev-packages] location in the Pipfile. These development packages only get installed if you specify the --dev argument with pipenv install.
```
Comment (GitLukeW) [inline README.md:289]: What's the [] here for?

**Review by GitLukeW — CHANGES_REQUESTED** [review]
Looks good, just some slight changes requested.

File: README.md (hunk 2 above)
Comment (esparr) [reply README.md:289]: That is how the section looks in the Pipfile. I feel like it is okay like this but if it caused a need for clarification, maybe I should update it to just say dev-package.

Comment (GitLukeW) [reply README.md:289]: Okay sounds good! I think that will be fine. I was just wondering if it was suppose to be a link that got missed.

**Comment by esparr** [comment]
> Looks good, just some slight changes requested.

Got this cleaned up. Should be good to re-review.

**Review by GitLukeW — APPROVED** [review]
Changes look good. Thank you!

**Pull request merged**